src/synapse/azext_synapse/custom.py

```diff
@@ -0,0 +1,178 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger  # pylint: disable=unused-import
+from knack.util import CLIError
+
+from azext_synapse.vendored_sdks.azure_synapse.models import ExtendedLivyBatchRequest, LivyStatementRequestBody, \
+    ExtendedLivySessionRequest
+
+from azext_synapse.vendored_sdks.azure_mgmt_synapse.models import Workspace, WorkspacePatchInfo, ManagedIdentity, \
+    DataLakeStorageAccountDetails, \
+    BigDataPoolResourceInfo, AutoScaleProperties, AutoPauseProperties, LibraryRequirements, NodeSize, NodeSizeFamily, \
+    SqlPool, Sku
+
+
+# pylint: disable=too-many-locals, too-many-branches, too-many-statements, unused-argument
+
+
+def list_spark_batch_jobs(cmd, client, workspace_name, spark_pool_name, from_index=None, size=None, detailed=True):
+    return client.list(workspace_name, spark_pool_name, from_index, size, detailed)
+
+
+def create_spark_batch_job(cmd, client, workspace_name, spark_pool_name, job_name, args, driver_memory, driver_cores,
+                           executor_memory, executor_cores, num_executors, job_type="SCALA",
+                           file=None, class_name=None, jars=None, files=None,
+                           archives=None, conf=None, artifact_id=None,
+                           tags=None, detailed=True):
+    dotnet_file = "local:///usr/hdp/current/spark2-client/jars/microsoft-spark.jar"
+    dotnet_class = "org.apache.spark.deploy.dotnet.DotnetRunner"
+
+    if job_type.upper() != "DOTNET" and (not file or not class_name):
+        raise CLIError('Scala and Python spark batch job must provide value for parameter file and class_name.'
+                       'If you want to create a DotNet spark batch job please add `--job-type DOTNET`.')
+
+    if job_type.upper() == "DOTNET":
+        file = dotnet_file
+        class_name = dotnet_class

```

Comment:
What if user provides `file` and `class_name` explicitly? Is it safe just to ignore user input? Or should we honor user input? Current logic chooses to ignore user input. Do you think we should notify users that for .NET spark, the `file` and `class_name` is non-adjustable?

Reply:
> What if user provides `file` and `class_name` explicitly? Is it safe just to ignore user input? Or should we honor user input? Current logic chooses to ignore user input. Do you think we should notify users that for .NET spark, the `file` and `class_name` are not able to change?

To keep consistent with Synapse studio, I think it is ok